src/app/routes/applications/governance-list/governance-list.component.ts

```diff
@@ -368,7 +370,27 @@ export class GovernanceListComponent implements OnInit, OnDestroy {
     this.dataSource.data = JSON.parse(JSON.stringify(this.origDatasource));
   }
 
+  createSubOrg(parentOrg: any) {
+    const dialogRef = this.dialog.open(NewOrganizationComponent, {
+      width: '600px',
+      data: {
+        viewType: ViewType.NEW,
+        parentOrg: JSON.parse(JSON.stringify(parentOrg)),
+        owner: parentOrg.owner
+      },
+      maxWidth: '100%',
+      disableClose: true
+    });
+
+    dialogRef.afterClosed()
+      .pipe(
+        filter(Boolean),
+        takeUntil(this.subscription$))
+      .subscribe(async () => await this.newSubOrg(parentOrg));
```

Comment:
This is probably due to my lack of RxJs knowledge but could you help me understand what this code is doing?

- In `pipe` you only take the not defined/not-null values (due to the `filter(Boolean)`). What happens if the return from `afterClosed()` is `undefined`?
- In `pipe`, `takeUntil(this.subscription$)` is a little unclear. Like I get that it means take until `this.subscription$` emits, but why should we do this? What is `this.subscription$` observing? Is this an Angular convention?
- `this.newSubOrg` is called on the `subscribe`. Does this mean that in theory `this.newSubOrg` could be called more than once?

Reply:
Ah now i remember. Filtering is needed because we can send there some values while closing dialogs. We are sending true/false values there. True when we want to refresh data and false if just closing a dialog.

Reply:
Ok, thanks for the explanations